test/unit/machine-learning/machine-learning-api-client.spec.ts

```diff
@@ -0,0 +1,159 @@
+    error: {
+      code: 404,
+      message: 'Requested entity not found',
+      status: 'NOT_FOUND',
+    },
+  };
+  const EXPECTED_HEADERS = {
+    'Authorization': 'Bearer mock-token',
+    'X-Firebase-Client': 'fire-admin-node/<XXX_SDK_VERSION_XXX>',
+  };
+  const noProjectId = 'Failed to determine project ID. Initialize the SDK with service '
+    + 'account credentials, or set project ID as an app option. Alternatively, set the '
+    + 'GOOGLE_CLOUD_PROJECT environment variable.';
+
+  const mockOptions = {
+    credential: new mocks.MockCredential(),
+    projectId: 'test-project',
+  };
+
+  const clientWithoutProjectId = new MachineLearningApiClient(
+    mocks.mockCredentialApp());
+
+  // Stubs used to simulate underlying api calls.
+  let stubs: sinon.SinonStub[] = [];
+  let app: FirebaseApp;
+  let apiClient: MachineLearningApiClient;
+
+  beforeEach(() => {
+    app = mocks.appWithOptions(mockOptions);
+    apiClient = new MachineLearningApiClient(app);
+  });
+
+  afterEach(() => {
+    _.forEach(stubs, (stub) => stub.restore());
+    stubs = [];
+    return app.delete();
+  });
+
+  describe('Constructor', () => {
+    it('should throw when the app is null', () => {
+      expect(() => new MachineLearningApiClient(null as unknown as FirebaseApp))
+        .to.throw('First argument passed to admin.machineLearning() must be a valid Firebase app');
+    });
+  });
+
+  describe('getModel', () => {
+    const INVALID_NAMES: any[] = [null, undefined, '', 1, true, {}, []];
+    INVALID_NAMES.forEach((invalidName) => {
+      it(`should reject when called with: ${JSON.stringify(invalidName)}`, () => {
+        return apiClient.getModel(invalidName)
+          .should.eventually.be.rejected.and.have.property(
+            'message', 'Model ID must be a non-empty string.');
+      });
+    });
+
+    it(`should reject when called with prefixed name`, () => {
+      return apiClient.getModel('projects/foo/models/bar')
+        .should.eventually.be.rejected.and.have.property(
+          'message', 'Model ID must not contain any "/" characters.');
+    });
+
+    it(`should reject when project id is not available`, () => {
+      return clientWithoutProjectId.getModel(MODEL_ID)
+        .should.eventually.be.rejectedWith(noProjectId);
+    });
+
+    it('should resolve with the requested model on success', () => {
+      const stub = sinon
+        .stub(HttpClient.prototype, 'send')
+        .resolves(utils.responseFrom({ name: 'bar' }));
+      stubs.push(stub);
+      return apiClient.getModel(MODEL_ID)
+        .then((resp) => {
+          expect(resp.name).to.equal('bar');
+          expect(stub).to.have.been.calledOnce.and.calledWith({
+            method: 'GET',
+            url: 'https://mlkit.googleapis.com/v1beta1/projects/test-project/models/1234567',
+            headers: EXPECTED_HEADERS,
+          });
+        });
+    });
+
+    it('should throw when a full platform error response is received', () => {
```

Comment:
`should reject`

Here and other similar tests.

Reply:
done